lib/robots/viomi/capabilities/ViomiGoToLocationCapability.js

```diff
@@ -0,0 +1,18 @@
+const GoToLocationCapability = require("../../../core/capabilities/GoToLocationCapability");
+const ViomiMapParser = require("../../../ViomiMapParser");
+const Logger = require("../../../Logger");
+
+class ViomiGoToLocationCapability extends GoToLocationCapability {
+    /**
+     * @param {import("../../../entities/core/ValetudoGoToLocation")} valetudoGoToLocation
+     * @returns {Promise<void>}
+     */
+    async goTo(valetudoGoToLocation) {
+        const point = ViomiMapParser.positionToViomi(valetudoGoToLocation.coordinates.x, valetudoGoToLocation.coordinates.y);
+        Logger.trace("location to go: ", point.x, point.y);
+        await this.robot.sendCommand("set_uploadmap", [0]);
+        await this.robot.sendCommand("set_pointclean", [1, point.x, point.y], {});
```

Comment:
Will this actually just move to the location or clean an area around it?

Reply:
It will go and clean square around 1.5x1.5 meter.

Reply:
In that case, it's the wrong capability, since GoTo should actually just go to a spot and idle there.

Instead, this would be spot cleanup.

I'm still not convinced that valetudo should implement that feature, because it seems to be just a worse version of the Zoned Cleanup with a predefined zone size.

Thoughts?

There's also #655 for the same thing on roborock

Reply:
I would like to have spot cleanup included, I use it a lot.

Reply:
If xiaomi/viomi has two separate functions "pointclean" and "zoneclean", then it should be different capabilities in valetudo.

Reply:
@aHVzY2g But is it any different from drawing a temporary zone around the spot you want to clean?

Reply:
I have to say no, but spot cleaning seems to work more reliable on the ViomiV7. If I draw a temporary zone the robot faces more than often some navigation issues.

Reply:
+1 spot cleaning without having to draw a zone is super useful.

Maybe also worth noting that Viomi also doesn’t have a go-to-xyz-and-do-nothing command (except maybe for go-to-charging-station).

Reply:
@rumpeltux Yeah but that doesn't necessarily require native support.
The UI could also provide an interface that works like that but actually draws the Zone for the Zoned Cleanup around the spot you're pointing at.

Reply:
> @aHVzY2g But is it any different from drawing a temporary zone around the spot you want to clean?

Yes there is.

If you draw an area and there's an obstacle in the middle it will bash on it for a while, as if it were going to jump out of existence if it bashes on it enough times.
If you do a "point clean" it will clean a circle centered on that point (as opposed to a square) and it won't bother too much about any obstacles.

It's most useful to clean a spot close to a wall, if your spot area covers the wall, with the "draw an area" mode it will take a lot longer than the "point clean" mode.